File: aulas/aula.js
// Aula de JavaScript - Conceitos Fundamentais

// 1. TIPOS DE DADOS BÁSICOS
let inteiro = 10;          // number
let real = 3.14;           // number
let texto = "Hello";       // string
let booleano = true;       // boolean

let lista = [1, 2, 3];     // array
let conjunto = new Set([1, 2, 2, 3]); // set
let dicionario = { nome: "Gui", idade: 17 }; // object

console.log(inteiro, real, texto, booleano);
console.log(lista, conjunto, dicionario);

// ------------------------------------------------------
// 2. OPERADORES
let x = 5, y = 2;
console.log(x + y);   // soma
console.log(x - y);   // subtração
console.log(x * y);   // multiplicação
console.log(x / y);   // divisão
console.log(x % y);   // resto
console.log(x ** y);  // potência

console.log(x == y, x > y, x !== y);

// ------------------------------------------------------
// 3. CONTROLE DE FLUXO
if (x > y) {
  console.log("x é maior que y");
} else if (x === y) {
  console.log("x é igual a y");
} else {
  console.log("x é menor que y");
}

// Loop for
for (let i = 0; i < 5; i++) {
  console.log("for:", i);
}

// Loop while
let contador = 0;
while (contador < 3) {
  console.log("while:", contador);
  contador++;
}

// ------------------------------------------------------
// 4. FUNÇÕES
function soma(a, b) {
  return a + b;
}
console.log("Função soma:", soma(3, 4));

// ------------------------------------------------------
// 5. ESTRUTURAS DE DADOS
let numeros = [10, 20, 30];
numeros.push(40);
console.log("Lista:", numeros);

let aluno = { nome: "Ana", nota: 9.5 };
console.log("Nome:", aluno.nome);

// ------------------------------------------------------
// 6. ENTRADA E SAÍDA
// let entrada = prompt("Digite algo:");
// console.log("Você digitou:", entrada);

// ------------------------------------------------------
// 7. TRATAMENTO DE ERROS
try {
  console.log(10 / 0);
} catch (e) {
  console.log("Erro:", e);
}

// ------------------------------------------------------
// 8. ORIENTAÇÃO A OBJETOS
class Pessoa {
  constructor(nome, idade) {
    this.nome = nome;
    this.idade = idade;
  }
  falar() {
    console.log(`Oi, eu sou ${this.nome} e tenho ${this.idade} anos.`);
  }
}

let p1 = new Pessoa("Gui", 17);
p1.falar();

// ------------------------------------------------------
// 9. MÓDULOS / BIBLIOTECAS
// Exemplo: usar a biblioteca Math
console.log("Raiz quadrada de 16:", Math.sqrt(16));

// ------------------------------------------------------
// 10. PARADIGMAS
// Imperativo
x = 10;
y = 20;
console.log(x + y);

// Funcional
let valores = [1, 2, 3, 4];
let dobro = valores.map(n => n * 2);
console.log("Dobro:", dobro);

// OO já vimos com a classe Pessoa

File: aulas/aula.py
# Aula de Python - Conceitos Fundamentais

# 1. TIPOS DE DADOS BÁSICOS
inteiro = 10              # int
real = 3.14               # float
texto = "Hello"           # string
booleano = True           # bool

lista = [1, 2, 3]         # list
conjunto = {1, 2, 2, 3}   # set (não aceita repetição)
dicionario = {"nome": "Gui", "idade": 17}  # dict

print(inteiro, real, texto, booleano)
print(lista, conjunto, dicionario)

# ------------------------------------------------------
# 2. OPERADORES
x, y = 5, 2
print(x + y)   # soma
print(x - y)   # subtração
print(x * y)   # multiplicação
print(x / y)   # divisão (float)
print(x // y)  # divisão inteira
print(x % y)   # resto
print(x ** y)  # potência

# Comparações retornam True/False
print(x == y, x > y, x != y)

# ------------------------------------------------------
# 3. CONTROLE DE FLUXO
if x > y:
    print("x é maior que y")
elif x == y:
    print("x é igual a y")
else:
    print("x é menor que y")

# Loop for
for i in range(5):
    print("for:", i)

# Loop while
contador = 0
while contador < 3:
    print("while:", contador)
    contador += 1

# ------------------------------------------------------
# 4. FUNÇÕES
def soma(a, b):
    return a + b

print("Função soma:", soma(3, 4))

# ------------------------------------------------------
# 5. ESTRUTURAS DE DADOS
numeros = [10, 20, 30]
numeros.append(40)
print("Lista:", numeros)

# Dicionário
aluno = {"nome": "Ana", "nota": 9.5}
print("Nome:", aluno["nome"])

# ------------------------------------------------------
# 6. ENTRADA E SAÍDA
# entrada = input("Digite algo: ")
# print("Você digitou:", entrada)

# ------------------------------------------------------
# 7. TRATAMENTO DE ERROS
try:
    print(10 / 0)
except ZeroDivisionError:
    print("Erro: divisão por zero")

# ------------------------------------------------------
# 8. ORIENTAÇÃO A OBJETOS
class Pessoa:
    def __init__(self, nome, idade):
        self.nome = nome
        self.idade = idade
    def falar(self):
        print(f"Oi, eu sou {self.nome} e tenho {self.idade} anos.")

p1 = Pessoa("Gui", 17)
p1.falar()

# ------------------------------------------------------
# 9. MÓDULOS / BIBLIOTECAS
import math
print("Raiz quadrada de 16:", math.sqrt(16))

# ------------------------------------------------------
# 10. PARADIGMAS
# Imperativo
x = 10
y = 20
print(x + y)

# Funcional (map, filter, reduce)
valores = [1, 2, 3, 4]
dobro = list(map(lambda n: n * 2, valores))
print("Dobro:", dobro)

# OO já vimos com a classe Pessoa
